configs/William/Msg_Map_process_test.py

```diff
@@ -0,0 +1,212 @@
+
+
+
+system.msg_queue.cpu_side = system.membus.mem_side_ports
+
+
+
+
+
+mem_ctrls = []
+
+num_chnls = 1
+intlv_bits = int(math.log(num_chnls, 2))
+cache_line_size = 64
+intlv_low_bit = int(math.log(cache_line_size, 2))
+
+for chnl in range(num_chnls):
+            # size = addr_range.size()
+            interface = HBM_1000_4H_1x128()
+            interface.range = AddrRange(addr_range.start, size = addr_range.size(), 
+                        intlvHighBit = intlv_low_bit + intlv_bits - 1,
+                        xorHighBit = 0,
+                        intlvBits = intlv_bits,
+                        intlvMatch = chnl)
+            ctrl = MemCtrl()
+            ctrl.dram = interface
+
+            #ctrl.dram.null = True
+            #ctrl.dram.addr_mapping = addr_map
+            #ctrl.dram.page_policy = page_policy
+            mem_ctrls.append(ctrl)
+
+my_ctrl = MemCtrl()
+my_ctrl.dram = DDR3_1600_8x8()
+my_ctrl.dram.range = queue_ranges[0] #system.mem_ranges[1]
+
+mem_ctrls.append(my_ctrl)
+
+# system.mem_ctrl.dram = DDR3_1600_8x8()
+# system.mem_ctrl.dram.range = system.mem_ranges[0]
+
+system.mem_ctrls = mem_ctrls
+
+# for mem_ctrl in system.mem_ctrls:
+#     mem_ctrl.port = system.membus.mem_side_ports
+system.mem_ctrls[0].port = system.membus.mem_side_ports
+# fake memory so thqat the queue is in the correct address range
+# system.msg_queue.mem_side_port = system.mem_ctrl2.port
+
+
+system.mem_ctrls[1].port = system.msg_queue.mem_side
+
+
+# # Connect the system up to the membus
+system.system_port = system.membus.cpu_side_ports
+
+# below here also for cpu
+# Create a process for a simple "Hello World" application
+process = Process()
+# process.map(vaddr=0x1000000, paddr=0x3CF9BDC0, size=4096, cacheable=False)
+# Set the command
+# grab the specific path to the binary
+thispath = os.path.dirname(os.path.realpath(__file__))
+
+binpath = os.path.join(thispath, '../../',
+                       'tests/test-progs/hello/bin/x86/linux/mapped_queue_fixed_private')
+# binpath = os.path.join(thispath, '../../',
+#                        'tests/test-progs/hello/bin/x86/linux/no_tuple')
+# cmd is a list which begins with the executable (like argv)
+process.cmd = [binpath]
+# Set the cpu to use the process as its workload and create thread contexts
+system.cpu.workload = process
+# system.cpu.workload.map(vaddr=0x1000000, paddr=0x3CF9BDC0, size=4096)
+system.cpu.createThreads()
+
+system.workload = SEWorkload.init_compatible(binpath)
+
+# set up the root SimObject and start the simulation
+root = Root(full_system = False, system = system)
+# instantiate all of the objects we've created above
+m5.instantiate()
+
+# process.map(vaddr=0x10000, paddr=0x3CF9BDC0, size=4096, cacheable=False)
+
+process.map(vaddr=0x1000000, paddr=0xFF00000, size=4096, cacheable=False)
+
+
+print("Beginning simulation!")
+exit_event = m5.simulate()
+print('Exiting @ tick %i because %s' % (m5.curTick(), exit_event.getCause()))
```

Comment:
Don't use `%` for formatting. Use [f-strings](https://peps.python.org/pep-0498/)

Another link: <https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python/#>